cg/services/deliver_files/file_formatter/component_files/concatenation_service.py

```diff
@@ -0,0 +1,307 @@
+    @staticmethod
+    def _get_concatenation_map(
+        forward_path: Path, reverse_path: Path, fastq_files: list[FastqFile]
+    ) -> dict[Path, Path]:
+        """
+        Get a list of ConcatenationMap objects for a sample.
+        NOTE: the fastq_files must be grouped by sample name.
+        args:
+            forward_path: Path: Path to the forward concatenated file.
+            reverse_path: Path: Path to the reverse concatenated file.
+            fastq_files: list[FastqFile]: List of fastq files for a single ample.
+        """
+        concatenation_map: dict[Path, Path] = {}
+        for fastq_file in fastq_files:
+            concatenation_map[fastq_file.fastq_file_path] = (
+                forward_path if fastq_file.read_direction == ReadDirection.FORWARD else reverse_path
+            )
+        return concatenation_map
+
+    @staticmethod
+    def _determine_read_direction(fastq_path: Path) -> ReadDirection:
+        """Determine the read direction of a fastq file.
+            Assumes that the fastq file path contains 'R1' or 'R2' to determine the read direction.
+        args:
+            fastq_path: Path: Path to the fastq file.
+        """
+        if f"R{ReadDirection.FORWARD}" in fastq_path.as_posix():
+            return ReadDirection.FORWARD
+        return ReadDirection.REVERSE
+
+    def _group_fastq_files_per_sample(
+        self, sample_names: set[str], fastq_files: list[FastqFile]
+    ) -> dict[str, list[FastqFile]]:
+        """Group fastq files per sample.
+        returns a dictionary with sample names as keys and a list of fastq files as values.
+        args:
+            sample_names: set[str]: Set of sample names.
+            fastq_files: list[FastqFile]: List of fastq files.
+        """
+
+        sample_fastq_files: dict[str, list[FastqFile]] = {
+            sample_name: [] for sample_name in sample_names
+        }
+        for fastq_file in fastq_files:
+            sample_fastq_files[fastq_file.sample_name].append(fastq_file)
+        self._all_sample_fastq_file_share_same_directory(sample_fastq_files=sample_fastq_files)
+        return sample_fastq_files
+
+    def _replace_fastq_paths(
+        self,
+        concatenation_maps: dict[Path, Path],
+        formatted_files: list[FormattedFile],
+    ) -> None:
+        """
+        Replace the fastq file paths with the new concatenated fastq file paths.
+        Uses the concatenation map with the formatted file path as key and the concatenated path as value.
+        args:
+            concatenation_maps: list[ConcatenationMap]: List of ConcatenationMap objects.
+            formatted_files: list[FormattedFile]: List of formatted files.
+        """
+        for formatted_file in formatted_files:
+            if self._is_lane_fastq_file(formatted_file.formatted_path):
+                formatted_file.formatted_path = concatenation_maps[formatted_file.formatted_path]
+
+    @staticmethod
+    def _all_sample_fastq_file_share_same_directory(
```

Comment:
The name of this method suggests a bool return, consider renaming to validate_all_fastq_files....